Port robot_state to ROS2

vmayoral opened the pull request:
Follows from https://github.com/ros-planning/moveit2/pull/13

vmayoral commented on moveit_core/robot_state/include/moveit/robot_state/log.h:
```diff
@@ -0,0 +1,53 @@
+{
+
+// Logger
+rclcpp::Logger logger_robot_state = rclcpp::get_logger("robot_state");
```
Follows previous code guidelines and creates a single logger `robot_state`.

vmayoral commented on moveit_core/robot_state/src/conversions.cpp:
```diff
@@ -39,24 +39,24 @@
 #include <geometric_shapes/shape_operations.h>
 #include <tf2_eigen/tf2_eigen.h>
 #include <boost/lexical_cast.hpp>
+#include <moveit/robot_state/log.h>
```
Uses logger created before. In case it's desired to create the logger here instead, please say it.

mlautman replied:
I agree with @henningkayser here. Adding a log.h each time we want a logger is overkill. see comment here: https://github.com/ros-planning/moveit2/pull/40#discussion_r264481130

henningkayser commented on moveit_core/robot_state/test/test_aabb.cpp:
```diff
@@ -150,32 +150,34 @@ TEST_F(TestAABB, TestPR2)
-  msg.type = visualization_msgs::Marker::CUBE;
-  msg.color.a = 0.5;
-  msg.lifetime.sec = 3000;
+  auto msg = std::make_shared<visualization_msgs::Marker>();
```
I'm not sure if it's always safe to reuse this shared pointer for multiple publish calls. Since we don't rely on performance in this test we might just pass the message by reference.

mlautman commented:
@anasarrak (Since the original PR was done by them)

vmayoral commented on moveit_core/robot_state/include/moveit/robot_state/robot_state.h:
```diff
@@ -950,114 +947,77 @@ as the new values that correspond to the group */
-  }
 
-  /** \brief  Warning: This function inefficiently copies all transforms around.
+  /** \brief  ing: This function inefficiently copies all transforms around.
```
c377881f6590a3134ae1f97e0e9d1664eb221fb4

vmayoral commented on moveit_core/robot_state/include/moveit/robot_state/robot_state.h:
```diff
@@ -950,114 +947,77 @@ as the new values that correspond to the group */
       @param constraint A state validity constraint to be required for IK solutions */
-  bool setFromIK(const JointModelGroup* group, const geometry_msgs::Pose& pose, double timeout = 0.0,
-                 const GroupStateValidityCallbackFn& constraint = GroupStateValidityCallbackFn(),
+  bool setFromIK(const JointModelGroup* group, const geometry_msgs::msg::Pose& pose, unsigned int attempts = 0,
```
@henningkayser can you help addressing this? what's your suggestion? is this something that will get fixed once we _sync_ with moveit1 master? If so, shall we skip your comment for now?

vmayoral replied:
Thanks @henningkayser, that makes sense. I recall @rhaschke pushing some stuff accidentally so it may be from that time since this PR is from a while ago.

I've created a new PR at https://github.com/ros-planning/moveit2/pull/59. Can you please have a look there?

vmayoral commented:
I've included @davetcoleman's approach from https://github.com/ros-planning/moveit2/issues/21 here. From @henningkayser's comments, it sounds like some sync is required to move forward with this PR. @henningkayser, you seem to be knowledgeable about what should be in or out in this submodule, can you help addressing your comments above?

vmayoral commented:
Follows at https://github.com/ros-planning/moveit2/pull/59.
